components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -524,16 +527,18 @@ private void redirectToAccountCreateUI(ExternalIdPConfig externalIdPConfig, Auth
             if (externalIdPConfig.isPasswordProvisioningEnabled()) {
                 uriBuilder.addParameter(FrameworkConstants.PASSWORD_PROVISION_ENABLED, String.valueOf(true));
             }
-            uriBuilder.addParameter(MultitenantConstants.TENANT_DOMAIN_HEADER_NAME, context.getTenantDomain());
+            if (!IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+                uriBuilder.addParameter(MultitenantConstants.TENANT_DOMAIN_HEADER_NAME, context.getTenantDomain());
+            }
             uriBuilder.addParameter(FrameworkConstants.SERVICE_PROVIDER, context.getSequenceConfig()
                     .getApplicationConfig().getApplicationName());
             uriBuilder.addParameter(FrameworkConstants.USERNAME, username);
             uriBuilder.addParameter(FrameworkConstants.SKIP_SIGN_UP_ENABLE_CHECK, String.valueOf(true));
             uriBuilder.addParameter(FrameworkConstants.SESSION_DATA_KEY, context.getContextIdentifier());
             addMissingClaims(uriBuilder, context);
             localClaimValues.forEach(uriBuilder::addParameter);
-            response.sendRedirect(uriBuilder.build().toString());
-        } catch (URISyntaxException | IOException e) {
+            response.sendRedirect(uriBuilder.build().getRelativeInternalURL());
```

Comment:
Don't we need to use `getRelativePublicURL` here?

Reply:
fixed